migrator/src/main/java/com/redhat/lightblue/migrator/ConsistencyCheckerController.java

```diff
@@ -0,0 +1,252 @@
+package com.redhat.lightblue.migrator;
+
+import java.util.Date;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.Update;
+import com.redhat.lightblue.client.Literal;
+import com.redhat.lightblue.client.request.data.DataInsertRequest;
+import com.redhat.lightblue.client.request.data.DataUpdateRequest;
+
+/**
+ * There is one consistency checker controller for each
+ * entity. Consistency checker controller is created and initialized
+ * from a MigrationConfiguration. The migration configuration gives
+ * the period with which the consistency checker runs. The consistency
+ * checker operates using a timestamp field. Every time it wakes up,
+ * it attempts to process documents that were created/modified after
+ * the last run time, and before (now-period).  When the consistency
+ * checker wakes up, it creates a migration job for every (period)
+ * length time slice.
+ *
+ * The consistency checker always leaves the last period
+ * unprocessed. This is to allow for any time discrepancies between
+ * nodes updating the database. If a node updates a record with a
+ * timestamp that is smaller than the current latest record, the
+ * consistency checker will miss that in the next run if the current
+ * maximum was used as the starting value of the next run.
+ */
+public class ConsistencyCheckerController extends AbstractController {
+
+    private static final Logger LOGGER=LoggerFactory.getLogger(ConsistencyCheckerController.class);
+
+    public ConsistencyCheckerController(Controller controller,MigrationConfiguration migrationConfiguration) {
+        super(controller,migrationConfiguration,"ConsistencyChecker:"+migrationConfiguration.getConfigurationName());
+    }
+
+    private static final class UOM {
+        final long multiplier;
+        final String[] names;
+
+        public UOM(long multiplier,String...names) {
+            this.names=names;
+            this.multiplier=multiplier;
+        }
+
+        public boolean in(String s) {
+            for(String n:names)
+                if(s.equalsIgnoreCase(n))
+                    return true;
+            return false;
+        }
+    }
+
+    private static final UOM[] uoms=new UOM[] {
+        new UOM(1l,"ms","msec","msecs"),
+        new UOM(1000l,"s","sec","secs","second","seconds"),
+        new UOM(60l*1000l,"m","min","mins","minute","minutes"),
+        new UOM(60l*60l*1000l,"h","hr","hrs","hour","hours"),
+        new UOM(24l*60l*60l*1000l,"d","day","days") };
+    
+    /**
+     * Returns the period in msecs
+     */
+    public static long parsePeriod(String periodStr) {
```

Comment:
Might be better to use yoda time library for parsing periods rather than maintaining that functionality in the migrator.